### ISSUE TITLE
[PoP Miner] 'mine' should occur on next BTC block detection, not cron.

### ISSUE THREAD

**Issue by gsxryan** [VeriBlock/nodecore-releases#101]
To decrease wasteful 'mine' commands and lost BTC, 'mine' should only occur when a new BTC block is detected, and when mempool has room.

Or, cron should only occur when mempool on current block has room.

**Comment by gsxryan** [comment]
Another way to describe;  Mine should occur with SAT under desired limits, and conditions are met to ensure a healthy PoP send.

**Comment by JoshRack** [comment]
We've introduced some new functionality in v0.3.6 that allows you to automate mining of specific VeriBlock blocks. I mention that as tangentially addresses your concern about "wasteful" mines. PoP mining is a competitive endeavor. A miner competes against other miners (and against other BTC transactions) for inclusion in as early a BTC block as possible. Each miner has to determine the cost/reward ratio they're comfortable with. The blockchain needs proof transactions on BTC to provide security, so I don't think we'd implement this feature as part of the core miner.

That said, we have discussed at length and plan to implement in the future an "opening up" of the miner to become more of a platform. The idea being to allow plugins where custom rules or logic could be applied. As you've articulated, one might decide they'd like the miner to shut down if they deem the market conditions "unfavorable".
